internal/controller/controllers/clusterdeployments_controller_test.go

```diff
@@ -1380,12 +1386,98 @@ var _ = Describe("cluster reconcile", func() {
+			mockInstallerInternal.EXPECT().UpdateClusterNonInteractive(gomock.Any(), gomock.Any()).Return(backEndCluster, nil).Times(3)
+			kubeconfig := `clusters:
+- name: foo
+  cluster:
+    server: "https://api.foo.bar:1234/baz"
+`
+			mockInstallerInternal.EXPECT().GetCredentialsInternal(gomock.Any(), gomock.Any()).Return(&models.Credentials{Password: "foo", Username: "bar"}, nil).Times(1)
+			mockInstallerInternal.EXPECT().V2DownloadClusterCredentialsInternal(gomock.Any(), gomock.Any()).Return(ioutil.NopCloser(strings.NewReader(kubeconfig)), int64(len(kubeconfig)), nil).Times(1)
+			mockInstallerInternal.EXPECT().HostWithCollectedLogsExists(gomock.Any()).Return(false, nil).Times(2)
+			mockClientFactory.EXPECT().CreateFromSecret(gomock.Any()).Return(mockClient, nil)
+			rootCAKey := types.NamespacedName{
+				Namespace: "kube-system",
+				Name:      "root-ca",
+			}
+			mockClient.EXPECT().Get(gomock.Any(), rootCAKey, gomock.AssignableToTypeOf(&corev1.ConfigMap{})).DoAndReturn(
+				func(ctx context.Context, name types.NamespacedName, configMap *corev1.ConfigMap) error {
+					configMap.Data = map[string]string{
+						MCS_CERT_NAME: "foobar",
+					}
+					return nil
+				},
+			)
+			request := newClusterDeploymentRequest(cluster)
+			result, err := cr.Reconcile(ctx, request)
+			Expect(err).To(BeNil())
 			Expect(result).To(Equal(ctrl.Result{}))
 			aci = getTestClusterInstall()
+			Expect(aci.Status.DebugInfo.State).To(Equal(models.ClusterStatusInstalled))
+
+			By("Reconcile to transform into day2 cluster")
+			day2backEndCluster := &common.Cluster{
+				Cluster: models.Cluster{
+					ID:                   backEndCluster.ID,
+					Name:                 clusterName,
+					OpenshiftVersion:     "4.8",
+					Status:               swag.String(models.ClusterStatusAddingHosts),
+					APIVip:               backEndCluster.APIVip,
+					BaseDNSDomain:        backEndCluster.BaseDNSDomain,
+					Kind:                 swag.String(models.ClusterKindAddHostsCluster),
+					APIVipDNSName:        swag.String(fmt.Sprintf("api.%s.%s", backEndCluster.Name, backEndCluster.BaseDNSDomain)),
+					HighAvailabilityMode: swag.String(models.ClusterHighAvailabilityModeNone),
+				},
+				PullSecret: testPullSecretVal,
+			}
+			mockInstallerInternal.EXPECT().TransformClusterToDay2Internal(gomock.Any(), gomock.Any()).Times(2).Return(day2backEndCluster, nil)
+			result, err = cr.Reconcile(ctx, request)
+			Expect(err).To(BeNil())
+			// First reconciliation will update Ignition URL and requeue updates
+			Expect(result).To(Equal(ctrl.Result{Requeue: true}))
+			result, err = cr.Reconcile(ctx, request)
+			Expect(err).To(BeNil())
+			Expect(result).To(Equal(ctrl.Result{Requeue: false}))
+			aci = getTestClusterInstall()
 			Expect(aci.Status.DebugInfo.State).To(Equal(models.ClusterStatusAddingHosts))
+			Expect(aci.Spec.IgnitionEndpoint).NotTo(BeNil())
+			ignitionEndpoint := fmt.Sprintf("https://1.2.3.5:22623/config/%s", hiveext.WorkerAgentMachinePool)
```

Comment:
As I see that, AgentClusterInstall as a whole will get ignition endpoint point to the **worker** ignition. Aren't we breaking day2 master with this approach?